compiler/rustc_traits/src/chalk/db.rs

```diff
@@ -247,12 +247,14 @@ impl<'tcx> chalk_solve::RustIrDatabase<RustInterner<'tcx>> for RustIrDatabase<'t
         };
         Arc::new(chalk_solve::rust_ir::FnDefDatum {
             id: fn_def_id,
-            abi: sig.abi(),
-            safety: match sig.unsafety() {
-                Unsafety::Normal => chalk_ir::Safety::Safe,
-                Unsafety::Unsafe => chalk_ir::Safety::Unsafe,
+            sig: chalk_ir::FnSig {
+                abi: sig.abi(),
+                safety: match sig.unsafety() {
+                    Unsafety::Normal => chalk_ir::Safety::Safe,
+                    Unsafety::Unsafe => chalk_ir::Safety::Unsafe,
+                },
+                variadic: sig.c_variadic(),
```

Comment:
We *could* add this `lowering` `impl<'tcx> LowerInto<chalk_ir::FngSig> for ty::FnSig<'tcx>`. Then this wouldn't be duplicated between here and below.

Reply:
Let me know if you want to do this. I'm fine either way.

Reply:
I implemented in on `ty::Binder<ty::FinSig<'tcx>>`, or should I implement on `ty::FnSig<'tcx>` and call `sig.skip_binder().lower_into(interner)`?

Reply:
It's fine as-is. Can always be changed in the future if things get changed.